### PR TITLE
feat(core): Notification events

### PR DESCRIPTION
**Pull request by NickSeagull**
## Description
<Describe the purpose of this pull request>

## Checks
- [x] Project Builds
- [x] Project passes tests and checks
- [x] Updated documentation accordingly

### CODE REVIEW

**Comment by NickSeagull** [comment]
Working on this for now

File: packages/framework-types/src/concepts/register.ts [hunk 1]
```diff
@@ -44,6 +45,15 @@ export class Register {
     return this
   }
 
+  /**
+   * Register a list of notifications to be added to the event-store on handler completion
+   * @param notifications
+   */
+  public notifications(...notifications: Array<NotificationInterface>): Register {
+    this.eventList.push(...notifications)
+    return this
```
Comment (javiertoledo) [inline packages/framework-types/src/concepts/register.ts:54]: Why not use the same `register.events` method to register notifications? It seems a bit confusing to me that we have a separate way to handle them when at the end of the day, they're also events... what do you think?

**Review by javiertoledo — CHANGES_REQUESTED** [review]
Great work here, and super-useful feature! I left you some suggestions; it could be worth considering minor renames in part of the code that handles both notification and state change events to make the code clearer. Also, I think there's a way to simplify how we store notifications metadata in the `Booster.config`. Let me know your thoughts!

File: packages/framework-core/src/decorators/notification.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        If you think that this is an error, try performing a clean build.`)
+      }
+      const topic = options?.topic ?? 'defaultTopic'
+      if (topic) {
+        config.eventToTopic[eventClass.name] = topic
+        config.topicToEvent[topic] = eventClass.name
+      }
+      config.notifications[eventClass.name] = {
+        class: eventClass,
+      }
```
[the remark this answers is not in the document]
Comment (NickSeagull) [reply packages/framework-core/src/decorators/notification.ts:29]: The first thing could lead to the fact that there's an event that's not a notification and has no entities associated and could trigger an error.

I left the notifications, and eventToTopic and topicToEvent separated on purpose in order to make this easier to separate in the future.

Notifications and entities would have their own fields, and the other two mappings would be part of the event store module.

**Pull request merged**